pkg/credentialprovider/config.go

```diff
@@ -202,20 +202,20 @@ func ReadURL(url string, client *http.Client, header *http.Header) (body []byte,
 	}
 	defer resp.Body.Close()
 
+	limitedReader := &io.LimitedReader{R: resp.Body, N: maxReadLength}
+	contents, err := ioutil.ReadAll(limitedReader)
+	if err != nil {
+		return nil, err
+	}
+
 	if resp.StatusCode != http.StatusOK {
-		klog.V(2).Infof("body of failing http response: %v", resp.Body)
+		klog.V(2).Infof("body of failing http response: %s", contents)
```

Comment:
I was doubt about the security of this code segment. As `ReadDockerConfigFileFromURL` will call this function, which may expose some auth info. correct me if i'm wrong.

Reply:
Hi, @kerthcet, this is a draft PR  for my first submit test.
What I consider is that if the request has failed, there should be no auth info, I don’t know if it's correct. 
waiting for sig/security to confirm #104184

Reply:
May be `InfoS` will be easy to resolve this https://github.com/kubernetes/kubernetes/issues/104184#issuecomment-893935406